appengine/flexible/websockets/templates/index.html

```diff
@@ -0,0 +1,98 @@
+{#
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#}
+{# [START index] #}
+<!doctype html>
+<html>
+  <head>
+    <title>Google App Engine Flexible Environment - Python Websockets Chat</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  </head>
+  <body>
+
+    <!-- [START form] -->
+    <p>Chat demo</p>
+    <form id="chat-form">
+      <textarea id="chat-text" placeholder="Enter some text..."></textarea>
+      <button type="submit">Send</button>
+    </form>
+
+    <div>
+      <p>Messages:</p>
+      <ul id="chat-response"></ul>
+    </div>
+
+    <div>
+      <p>Status:</p>
+      <ul id="chat-status"></ul>
+    </div>
+    <!-- [END form] -->
+
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+    <script>
+    // [START js]
+    $(function() {
+      /* If the main page is served via https, the WebSocket must be served via
+         "wss" (WebSocket Secure) */
+      var scheme = window.location.protocol == "https:" ? 'wss://' : 'ws://';
+      var webSocketUri =  scheme
+                          + window.location.hostname
+                          + (location.port ? ':'+location.port: '')
+                          + '/chat';
+
+      /* Get elements from the page */
+      var form = $('#chat-form');
+      var textarea = $('#chat-text');
```

Comment:
Nit: should this be moved into an included `.js` file, or is that too much work?

Reply:
For simplicity's sake (managing statics is another layer of complexity beyond templates) I'd say it's clearer as-is